DependencyInjection/Compiler/LoggerPass.php

```diff
@@ -28,10 +29,9 @@ public function process(ContainerBuilder $container)
             return;
         }
 
-        $subscriber = $container->getDefinition('fos_http_cache.event_listener.log')
-            ->setAbstract(false);
-
-        $container->getDefinition('fos_http_cache.cache_manager')
-            ->addMethodCall('addSubscriber', array($subscriber));
+        $container->getDefinition('fos_http_cache.event_listener.log')
+            ->setAbstract(false)
+            ->addTag('kernel.event_subscriber');
```

Comment:
why did we not do things like this previously? and should we instead just make the logger an optional dependency and have the subscriber not subscribe to anything when logger is null?